### PR TITLE
Refactors screwdriver coloring to use greyscale

### PR DESCRIPTION
**Pull request by ghost**
While working on #29180, it was brought up, quite astutely, that screwdrivers' colors are poorly handled. Each variety of screwdriver handle had its own sprite, resulting in around 7-odd sprites just for any color. Because this is silly, I decided to fix it!

Screwdrivers' handles now have their own icon state, and it's greyscale. Upon generation, the handle is given one of their seven old colors and the screwdriver's head and handle highlights are applied, with no color. In addition to providing a huge variety of possible colors for your fancy new screwdriver, this removes a lot of the clutter in `tools.dmi`.

Screwdrivers will not generate colors if their icon state is set to anything other than the handle when they're created. This is so that screwdrivers like the brass screwdriver and hand drill aren't modified at all and can keep their unique look.

### CODE REVIEW

**Comment by BeeSting12** [comment]
love the branch name. I believe wirecutters work similarly if you feel like doing these. This is pretty cool, can't wait to have a rainbow of screwdrivers!

**Comment by banthebantz** [comment]
gotta catch em all

**Comment by Xhuis** [comment]
I'm planning on doing wirecutters later today.

File: code/game/objects/items/weapons/tools.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/tools.dm]: don't check icon_state for logic

File: code/game/objects/items/weapons/tools.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/tools.dm]: don't check icon_state for logic

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/tools.dm
Comment (Xhuis) [inline code/game/objects/items/weapons/tools.dm]: What do you think I should use? A var like `random_colors` of some sort?

**Comment by ChangelingRain** [comment]
What about inhand states.

File: code/game/objects/items/weapons/tools.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/tools.dm]: yeah just something unique to the object

**Comment by Xhuis** [comment]
Screwdrivers don't have unique inhands, you doofus.

**Comment by Xhuis** [comment]
Huh. It looks like there's something weird about screwdriver inhands. They worked during testing, but I'm not sure of the specifics so I'll mess around with 'em a bit.

Sorry for calling you a doofus! I'm the real doofus.

**Comment by Xhuis** [comment]
@WJohn As the art director, you have final say on this. What're your thoughts on this?

**Comment by RandomMarine** [comment]
Can this cause confusion with syndie screwdrivers?

**Comment by Xhuis** [comment]
Very unlikely

**Comment by Xhuis** [comment]
Due to several people making very good points about the new colors, I've shrunk the list of possible colors down to the seven old ones, but also kept in place the new framework. This makes it faster to assign the colors to them but keeps the old look and cuts down on .dmi bloat.

Inhands are still admittedly a WIP.

**Comment by dannno** [comment]
Not sure this falls under art direction to be honest
the sprites are the same as always, just colored smarter

**Comment by Xhuis** [comment]
Before I added that, the colors were random in the RGB spectrum. That also explains the comments from before. Some maintainers didn't like that, so I consulted WJ and he helped me out with getting the old colors, just applying them well.

**Comment by Xhuis** [comment]
Inhands now work properly. Each color in the list is now the association of a string (i.e. "blue", "green"...), and when a color is chosen randomly, the `item_state` is set based on that string. The color is then applied through the association.

**tl;dr** I did the thing, and this doesn't change colors anymore, just makes them more efficiently applied and reduces bloat.

**Review by ChangelingRain — DISMISSED** [review]
You need dynamic inhand colors.

**Comment by Xhuis** [comment]
I'm currently in the process of making those!

**Comment by Xhuis** [comment]
Both of these inhands are dynamically generated!

That's dynamic inhands, now done. Since @ChangelingRain wanted a comparison image of the colors between old and new, let me get that.

**Comment by MoreRobustThanYou** [comment]
@Xhuis do this for cable coils too, please

**Comment by Xhuis** [comment]
That's something that's super easy to do, but I'm not gonna do it for this PR due to atomization.

File: code/game/objects/items/weapons/tools.dm
Comment (MrPerson) [inline code/game/objects/items/weapons/tools.dm]: Just a reminder that this exists

**Review by MrPerson — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/tools.dm
Comment (Xhuis) [inline code/game/objects/items/weapons/tools.dm]: Frick

**Comment by Xhuis** [comment]
# New

# Old


The new ones look about the same but are a little easier on the eyes, at least to me.

**Comment by ChangelingRain** [comment]
Those are much darker and should not change appearance at all.

**Comment by ExcessiveUseOfCobblestone** [comment]
why not RAINBOW SCREWDRIVERS 🌈

File: code/__HELPERS/unsorted.dm
Comment (nicbn) [inline code/__HELPERS/unsorted.dm]: This proc is never called

**Review by nicbn — CHANGES_REQUESTED** [review]

**Comment by nicbn** [comment]
Why don't you just random up the hue and keep the saturation and brightness the same? That way it will look random but not oversaturated.

**Comment by Xhuis** [comment]
The answer to both of those questions is because two maintainers want the appearances to remain exactly the same as the old ones, down to the pixel.

**Comment by Xhuis** [comment]
Alright, this should be the last commit of mine.

@WJohn As the Sprite Man I figured I should ask you, as @ChangelingRain and I were having some disagreements on it. Are the new greyscaled sprites similar-looking enough to be worth the code changes?

**Comment by optimumtact** [comment]
I think this looks fine and reduces sprites ! :+1:

**Comment by Cheridan** [comment]
its fine

**Comment by Jalleo** [comment]
Headmin overrule merge?

File: code/game/objects/items/weapons/tools.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/tools.dm]: This will suffer from global name collision, name it more uniquely

File: code/game/objects/items/weapons/tools.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/tools.dm]: This var is never used?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by nicbn** [comment]
There still are merge conflicts, you need to add the synthesizer sprites to inhands dmis. #29197

**Comment by Xhuis** [comment]
Dangit!

File: code/modules/mob/living/carbon/human/update_icons.dm
Comment (nicbn) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Is this intentional?

File: code/modules/mob/living/carbon/update_icons.dm
Comment (nicbn) [inline code/modules/mob/living/carbon/update_icons.dm]: Same for here

File: code/modules/mob/living/carbon/human/update_icons.dm
Comment (Xhuis) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Yes

File: code/modules/mob/living/carbon/update_icons.dm
Comment (Xhuis) [inline code/modules/mob/living/carbon/update_icons.dm]: Also yes

File: code/modules/mob/living/carbon/human/update_icons.dm
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Yes, he uses it for screwdrivers

File: code/modules/mob/living/carbon/update_icons.dm
Comment (RemieRichards) [inline code/modules/mob/living/carbon/update_icons.dm]: See other comment.

**Comment by WJohn** [comment]
I already said I was okay with this in coderbus but I guess I have to give the okay here, too.

so.

👌

**Comment by Xhuis** [comment]
This should be ready, then

**Review by nicbn — APPROVED** [review]

**Pull request merged**
